### PR TITLE
Fix for Issue related to importing a set module

### LINKED ISSUE #1987: Handling of new non-JS module types: JSON, CSS, HTML
Is there a plan for handling non-JS module types? As JSON, CSS, and HTML modules become standardized and implemented, I think it'd be very useful for SystemJS to allow for `import styles from './styles.css'`, etc.

### PR DESCRIPTION
**Pull request by TheMcMurder**
Fixes https://github.com/systemjs/systemjs/issues/1927

### CODE REVIEW

File: dist/extras/named-register.js [hunk 1]
```diff
@@ -40,4 +40,4 @@
   systemJSPrototype.instantiate = function (url, firstParentUrl) {
     return this.registerRegistry[url] || instantiate.call(this, url, firstParentUrl);
   };
-})();
\ No newline at end of file
+})();
```
Comment (joeldenning) [inline dist/extras/named-register.js]: The dist directory is in version control, but you shouldn't push changes to it in PRs -- Guy Bedford runs the build right before publishing and that's the only time the dist directory files should change.

File: src/features/registry.js [hunk 2]
```diff
@@ -79,3 +80,11 @@ systemJSPrototype.entries = function () {
     [iterator]: function() { return this }
   };
 };
+
+systemJSPrototype.resolve = function (id, parentUrl) {
```
Comment (joeldenning) [inline src/features/registry.js]: Out of curiosity, why did you move this from import-map.js to registry.js?

File: dist/extras/named-register.js (hunk 1 above)
Comment (TheMcMurder) [reply dist/extras/named-register.js]: That makes Waaaay more sense. I'll update that tonight. I didn't add (or try to add) any tests yet, so I'd also like to look into that

File: src/features/registry.js (hunk 2 above)
Comment (TheMcMurder) [reply src/features/registry.js]: Good question. It was somewhat arbitrary, I was trying to keep logic related to the registry/import map in their respective files. I wasn't really happy with it but it _felt_ slightly better than doing the check in the import-map file.

I'm happy to change it if the behavior I identified is actually a bug and worth fixing.

Comment (joeldenning) [reply src/features/registry.js]: 👍 I don't think there's a bug with it -- my understanding is that system-core.js is a barebones implementation that requires something else to implement a `resolve()` function for it. The risk of moving the resolve function from one file to another would be that not all builds of systemjs include the same files (see [s.js](https://github.com/systemjs/systemjs/blob/master/src/s.js) and [system.js](https://github.com/systemjs/systemjs/blob/master/src/system.js), and not all builds support import-maps.

The `s.js` small build doesn't support import maps, but it also doesn't include `registry.js` as part of its build. So I think all places that use `registry.js` now use import maps and that this won't cause any bugs.

Will defer of course to Guy Bedford for the more definitive answer on this

**Comment by guybedford** [comment]
Thanks for the PR.

See https://github.com/systemjs/systemjs/issues/1927#issuecomment-482815118 for the feedback discussion here.

**Comment by TheMcMurder** [comment]
Awesome! Thanks for helping me understand. Closing this PR because it isn't necessary.
